org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/uriresolver/XMLCacheResolverExtension.java

```diff
@@ -32,6 +33,30 @@
  */
 public class XMLCacheResolverExtension implements URIResolverExtension {
 
+	private static class XMLFileInputSource extends XMLInputSource {
+
+		private final Path file;
+
+		public XMLFileInputSource(XMLResourceIdentifier resourceIdentifier, Path file) {
+			super(resourceIdentifier);
+			this.file = file;
+		}
+
+		@Override
+		public InputStream getByteStream() {
+			// Load the file input stream only if it is used
+			InputStream input = super.getByteStream();
+			if (input == null) {

```

Comment:
check if path exists and is a file

Reply:
This test is done before (file is null when it doesn't exists), in otherwise don't need to do that here.